qoqo/src/operations/multi_qubit_gate_operations.rs

```diff
@@ -64,3 +64,306 @@ pub struct MultiQubitZZ {
+                let mut a = vec![];
+                for param in free_parameters {
+                    a.push(convert_into_calculator_float(param.bind(py)).map_err(|_| {
+                        pyo3::exceptions::PyTypeError::new_err(
+                            "Argument gate time cannot be converted to CalculatorFloat",
+                        )
+                    })?)
+                }
+                Ok(a)
+            })?;
+
+        Ok(Self {
+            internal: CallDefinedGate::new(gate_name, qubits, free_parameters_cf),
+        })
+    }
+
+    /// Return the name of the gate to apply.
+    ///
+    /// Returns:
+    ///     String: The name of the gate.
+    fn gate_name(&self) -> String {
+        self.internal.gate_name().clone()
+    }
+
+    /// Return the qubits on which the Gate operation is applied.
+    ///
+    /// Returns:
+    ///     Vec<int>: The qubits of the operation.
+    fn qubits(&self) -> Vec<usize> {
+        self.internal.qubits().clone()
+    }
+
+    /// Return the qubits on which the Gate operation is applied.
+    ///
+    /// Returns:
+    ///     Vec<CalculatorFloat>: The qubits of the operation.
+    fn free_parameters(&self) -> Vec<CalculatorFloatWrapper> {
+        self.internal
+            .free_parameters()
+            .iter()
+            .map(|param| CalculatorFloatWrapper {
+                internal: param.clone(),
+            })
+            .collect::<Vec<CalculatorFloatWrapper>>()
+    }
+
+    /// List all involved qubits.
+    ///
+    /// Returns:
+    ///     set[int]: The involved qubits of the operation.
+    fn involved_qubits(&self) -> PyObject {
+        let pyobject: PyObject = Python::with_gil(|py| -> PyObject {
+            PySet::new_bound(py, &[self.internal.qubits().clone()])
+                .unwrap()
+                .to_object(py)
+        });
+        pyobject
+    }
+
+    /// Return tags classifying the type of the operation.
+    ///
+    /// Used for the type based dispatch in ffi interfaces.
+    ///
+    /// Returns:
+    ///     list[str]: The tags of the Operation.
+    fn tags(&self) -> Vec<String> {
+        self.internal.tags().iter().map(|s| s.to_string()).collect()
+    }
+
+    /// Return hqslang name of the operation.
+    ///
+    /// Returns:
+    ///     str: The hqslang name of the operation.
+    fn hqslang(&self) -> &'static str {
+        self.internal.hqslang()
+    }
+
+    /// Return true when the operation has symbolic parameters.
+    ///
+    /// Returns:
+    ///     is_parametrized (bool): True if the operation contains symbolic parameters, False if it does not.
+    fn is_parametrized(&self) -> bool {
+        self.internal.is_parametrized()
+    }
+
+    /// Substitute the symbolic parameters in a clone of the operation according to the input.
+    ///
+    /// Args:
+    ///     substitution_parameters (dict[str, float]): The dictionary containing the substitutions to use in the operation.
```

Comment:
I have no preference between using Dict vs dict, but you are using `dict` here and `Dict` style in your other new operation (though there it is a `List`). Please standardise these, whichever way you prefer